openupgrade_scripts/scripts/l10n_vn/17.0.2.0.3/pre-migration.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024 Viindoo Technology Joint Stock Company (Viindoo)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from openupgradelib import openupgrade
+
+_account_tax_group_xmlid = [
+    "l10n_vn.tax_group_0",
+    "l10n_vn.tax_group_10",
+    "l10n_vn.tax_group_5",
+]
+
+_account_tax_xmlid = [
+    "l10n_vn.tax_purchase_vat0",
+    "l10n_vn.tax_purchase_vat10",
+    "l10n_vn.tax_purchase_vat5",
+    "l10n_vn.tax_sale_vat0",
+    "l10n_vn.tax_sale_vat10",
+    "l10n_vn.tax_sale_vat5",
+]
+
+
+def _vn_coa_rename_xml_id(env):
+    """
+    Since the removal of account.chart.template
+    we need to rename some xml_id like tax or tax.group
+    in order to avoid duplication
+    """
+    env.cr.execute("""SELECT id, name FROM res_company WHERE chart_template = 'vn'""")
+    xmlids_renames = []
+    for company_id, _ in env.cr.fetchall():
+        for tax_group_xmlid in _account_tax_group_xmlid:
+            new_xmlid = f"account.{company_id}_" + tax_group_xmlid.split(".")[1]
+            xmlids_renames.append((tax_group_xmlid, new_xmlid))
```

Comment:
@duong77476-viindoo có 2 công ty cùng dùng COA vn thì sao nhỉ?
16.0 đang dùng chung tax group giữa các công ty, anh nghĩ lên 17 sẽ cần xử như sau:
tax group sẽ đổi xml với tiền tố công ty đầu tiên, nếu có các công ty khác nữa thì phải nhân bản lên với xmlid tương ứng

account tax cũng tương tự

Reply:
E có for company trên rồi mà anh, 2 công ty Vn thì 1 cty id là 2 1 cty với id là 3 thì vẫn đúng mà ta

Reply:
@duong77476-viindoo em đang dùng rename xml, thì chỉ có 1xml mà em renamed thành 2 cái thì đâu có đúng đâu?
trừ khi rename không có thì tạo mới